packages/common/src/utils/common.ts

```diff
@@ -1,17 +1,21 @@
 import type { Playlist, PlaylistItem } from '../../types/playlist';
 
 export function debounce<T extends (...args: any[]) => void>(callback: T, wait = 200) {
-  let timeout: NodeJS.Timeout | null;
-  return (...args: unknown[]) => {
+  let timeout: NodeJS.Timeout | undefined;
+  function debounced(...args: unknown[]) {
     if (timeout) clearTimeout(timeout);
     timeout = setTimeout(() => callback(...args), wait);
-  };
+  }
+
+  debounced.cancel = () => clearTimeout(timeout);
```

Comment:
Any reference implementation you plan to add?

Reply:
Yes, in LayoutGrid.tsx:

```tsx
    // cancel the scrollIntoView action when the focus leaves the layout grid
    if (isLeaving) scrollIntoViewThrottled.cancel();
```

and:

```tsx
useEffect(() => {
  return () => scrollIntoViewThrottled.cancel();
}, []);
```